graph/case_similarity.py

```diff
@@ -28,6 +29,21 @@ def most_similar_to_group(self, opinion_ids: set) -> Dict[str, float]:
                 / len(opinion_neighbor_sets)
         return similarity_value_dict
 
+    def internal_similarity(self, opinion_ids: set) -> Dict[frozenset, float]:
+        """
+        Returns the internal similarity relationships in a group of cases.
+        """
+        neighbor_dict = {oi: set(self.network.neighbors(oi)) for oi in opinion_ids}
+        output_graph = nx.Graph()
+
+        for id_ in opinion_ids:
+            output_graph.add_node(int(id_))
+
+        for id_1, id_2 in itertools.combinations(opinion_ids, 2):
+            output_graph.add_edge(id_1, id_2, weight = \
+                self.jaccard_index(neighbor_dict[id_1], neighbor_dict[id_2]))
```

Comment:
It might be a little simpler to just networkx's complete_graph helper, which automatically creates a new graph with edges between all provided nodes. Something like this: (disclaimer: haven't tried this)
```suggestion
        output_graph = nx.complete_graph(opinion_ids)  # Creates a graph with those nodes and edges between all nodes

        for id_1, id_2 in output_graph.edges:
            output_graph[id_1][id_2]['weight'] = \
                self.jaccard_index(neighbor_dict[id_1], neighbor_dict[id_2]))
```